versions/draft-3/SPEC.md

```diff
@@ -994,6 +995,40 @@ task docker_test {
 }
 ```
 
+### Hints Section
+
+Hints are specified in a `hints` section using the same key/value format as the `runtime` section:
+
+```wdl
+task test {
+  command {
+    python script.py
+  }
+  runtime {
+    docker: "ubuntu:latest"
+  }
+  hints {
+    memory: "256MB"
+  }
+}
+```
+
+
+Hints allow a WDL author to provide recommendations to the engine when running a task. 
+
+* Unlike `runtime` these fields are not specified to indicate *what* a workflow does, but rather to provide hints to the engine about *how* to run a workflow.
+* Also unlike `runtime`, the `hints` section allows arbitrary keys and values beyond the set defined below to allow platform specific collaboration between WDL author and platform.
+* The only requirement on new `hints` from the spec is that no engine may implement a `hint` which materially affects the outcome of the task. These should instead be handled as spec changes for additions to the `runtime` section.
+* It is intended that over time the number of fields that an author places in the `hints` section decreases as workflow-submission APIs mature to define the same set of options in a more appropriate abstraction layer.
+
+The following `hints` are reserved by the WDL spec and have a known format. An engine is free to ignore them, but must not override them with an alternative format and implementation. Because hints do not materially affect workflow outcomes, they have no fixed default values. An engine is free to implement whatever defaults it likes:
+
+|Key|Type|Description|
+|---|---|---|
+|`memory`|`String` or `Int`|See [memory](#memory) section below.|
+|`cpu`|`Int`|A recommended minimum number of CPUs for the task. |
+
```

Comment:
The `disk` property is missing here. Is this an intentional omission?

Reply:
Nope - just an oversight (and more evidence that a single place to record all these will be useful!)